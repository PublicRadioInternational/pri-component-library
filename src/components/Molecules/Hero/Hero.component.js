/**
 * @file Hero.component.js
 * Exports a Hero component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Hero.css';

import Icon from '../../Atoms/Svg/Icons.component';
import LazyLoad from '../../Atoms/LazyLoad/LazyLoad.component';

const Hero = ({
  title,
  summary,
  hasAudio,
  category,
  categoryUrl,
  imgSrc,
  imgAlt,
  url
}) => (
  <article typeof="sioc:Item foaf:Document" className={styles.hero}>
    <figure className={styles.figure}>
      <LazyLoad>
        <img className={styles.img} data-src={imgSrc} alt={imgAlt} />
      </LazyLoad>
    </figure>
    <div className={styles.text}>
      {hasAudio && (
        <a className={styles.iconLink} href={url}>
          <Icon
            name="volume"
            className={styles.icon}
            isRoundIcon
            ariaLabel="Audio"
          />
        </a>
      )}
      <h2>
        <a className={styles.title} href={url}>
          {title}
        </a>
      </h2>
      <p className={styles.summary}>{summary}</p>
      <a className={styles.category} href={categoryUrl || url}>
        {category}
      </a>
    </div>
  </article>
);

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  categoryUrl: PropTypes.string,
  imgSrc: PropTypes.string.isRequired,
  imgAlt: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  hasAudio: PropTypes.bool
};

Hero.defaultProps = {
  hasAudio: false,
  categoryUrl: null
};

export default Hero;
