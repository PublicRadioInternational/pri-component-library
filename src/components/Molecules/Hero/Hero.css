/* import colors... */
@value colors: "../../00_global/colors.css";
@value gray, orange, grayLighter from colors;

/* import breakpoints */
@value small as bp-small, medium as bp-medium, header as bp-header from "../../00_global/breakpoints.css";

.hero {
  position: relative;
  overflow: hidden;
  background: black;
  border-bottom: 1px solid gray;
  margin-bottom: 1.5rem;
  color: #ddd;
  text-align: center;
  text-shadow: 1px 0px 20px gray;
}

.figure {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  margin: 0;
  opacity: 0.8;
}

.figure::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.5;
  background: linear-gradient(to bottom, rgba(0, 0, 0, 0) 0%, black 100%);
  box-shadow: inset 0px 15px 100px 0px black;
}

.img {
  height: 100%;
  object-fit: cover;
  width: 100%;
}

.text {
  position: relative;
  width: 93%;
  max-width: 800px;
  margin: 0 auto;
  padding: 175px 15px 30px;
  font-size: 1rem;
  text-shadow: 0 1px 0 #000;
}

@media bp-small {
  .text {
    max-width: 540px;
  }
}

@media bp-medium {
  .text {
    max-width: 720px;
  }
}

@media bp-header {
  .text {
    max-width: 720px;
  }
}

.iconLink {
  color: white;
  display: block;
  margin-bottom: 0.5rem;
}

.iconLink:hover {
  color: grayLighter;
}

.icon {
  height: 30px;
  padding: 1px 1px 0;
  width: 30px;
}

.title {
  color: #ddd;
  display: block;
  margin-bottom: 1rem;
  text-decoration: none;
}

.title:hover {
  color: white;
}

.summary {
  font-size: 1.2rem;
}

.category {
  color: #ddd;
  font-weight: bold;
  text-decoration: none;
}

.category:hover {
  color: white;
}
