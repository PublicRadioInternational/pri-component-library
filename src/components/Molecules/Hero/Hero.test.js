/**
 * @file hero
 * Contains tests for Hero.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Hero from './Hero.component';

jest.mock('intersection-observer');

describe('<Hero />', () => {
  it('Matches the Hero snapshot', () => {
    const component = renderer
      .create(
        <Hero
          hasAudio
          url="https://google.com"
          title="Opioid addicts are turning to a clinic in Mexico to break their addiction, but there's a catch. The treatment is illegal in the US."
          summary="Ibogaine is a powerful hallucinogenic drug that has been shown to have surprising success in treating addiction. But it comes with risks that, so far, have made its use in the US illegal."
          category="GlobalPost"
          categoryUrl="https://bing.com"
          imgSrc="https://media.pri.org/s3fs-public/styles/feature_large/public/story/images/JL1_8064.JPG?itok=WfGYKL2U"
          imgAlt="Opioid Addiction"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
