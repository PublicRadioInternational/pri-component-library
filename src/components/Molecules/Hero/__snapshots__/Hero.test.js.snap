// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> Matches the Hero snapshot 1`] = `
<article
  className="hero"
  typeof="sioc:Item foaf:Document"
>
  <figure
    className="figure"
  >
    <img
      alt="Opioid Addiction"
      className="img"
      data-src="https://media.pri.org/s3fs-public/styles/feature_large/public/story/images/JL1_8064.JPG?itok=WfGYKL2U"
    />
  </figure>
  <div
    className="text"
  >
    <a
      className="iconLink"
      href="https://google.com"
    >
      <svg
        aria-hidden={null}
        aria-label="Audio"
        aria-labelledby={null}
        className="svg icon roundIcon"
        fill="currentcolor"
        height={28}
        version={1.1}
        viewBox="0 0 28 28"
        width={28}
      >
        <path
          d="M9 10h-4v6h4l5 5v-16l-5 5zM20.36 19.36l-1.41-1.41c1.272-1.265 2.059-3.015 2.059-4.95s-0.787-3.685-2.059-4.95l-0-0 1.41-1.41c1.634 1.625 2.645 3.874 2.645 6.36s-1.011 4.735-2.644 6.36l-0 0zM17.54 16.54l-1.42-1.42c0.542-0.543 0.877-1.292 0.877-2.12s-0.335-1.577-0.877-2.12l0 0 1.42-1.42c0.912 0.903 1.477 2.156 1.477 3.54s-0.565 2.637-1.477 3.54l-0 0z"
        />
      </svg>
    </a>
    <h2>
      <a
        className="title"
        href="https://google.com"
      >
        Opioid addicts are turning to a clinic in Mexico to break their addiction, but there's a catch. The treatment is illegal in the US.
      </a>
    </h2>
    <p
      className="summary"
    >
      Ibogaine is a powerful hallucinogenic drug that has been shown to have surprising success in treating addiction. But it comes with risks that, so far, have made its use in the US illegal.
    </p>
    <a
      className="category"
      href="https://bing.com"
    >
      GlobalPost
    </a>
  </div>
</article>
`;
